src/asana/controller.py

```diff
@@ -33,11 +34,28 @@ def update_task(pull_request: PullRequest, task_id: str):
         for k, v in fields.items()
         if k in ("assignee", "name", "html_notes", "completed", "custom_fields")
     }
+    task = asana_client.get_task(task_id)
+    new_due_on = _new_due_on_or_none(task, update_task_fields)
+    if new_due_on is not None:
+        update_task_fields["due_on"] = new_due_on
     asana_client.update_task(task_id, update_task_fields)
     asana_client.add_followers(task_id, fields["followers"])
     maybe_complete_tasks_on_merge(pull_request)
 
 
+def _new_due_on_or_none(task: dict, update_task_fields: dict) -> Optional[str]:
+    due_on = None
+    default_due_on = asana_helpers.default_due_date_str()
+
+    if task["due_on"] > default_due_on:
+        # don't update due dates that were potentially manually updated past the default
+        return None
+    elif (task.get("assignee") or {}).get("gid") != update_task_fields.get("assignee"):
+        # if the task is switching assignees, update the due date to today
+        return datetime.now().strftime("%Y-%m-%d")
```

Comment:
If someone redirects a PR to someone else because they weren't the right person to receive it, it seems like that would update the due date to today, which seems aggressive. What if all this logic instead encoded:
* If assignee changes, update the due date to today.
* except: never update the due date to an earlier time

Reply:
Yeah I thought about that (which would basically switch the logic above to return None if the _existing_ due date is tomorrow or later, rather than just later than tomorrow) - but I thought it was more common that someone reviews same-day, which then should re-assign to today for next steps.

However, now thinking about it more - I can see a common use case where someone opens a PR assigned to themself, then self-reviews and cleans up a bit, then assigns it to a peer. In that case, we'd want to keep the "next business day" logic.

I think you're right - I can start with a less aggressive approach and see if feedback comes in at all.

Reply:
If it's easy to do, giving this logic awareness of the PR's author (so it can compare to the new assignee) could also allow it to be more targeted.